File: src/main/java/toast/client/commands/cmds/Panic.java
package toast.client.commands.cmds;

import toast.client.ToastClient;
import toast.client.commands.Command;
import toast.client.modules.Module;

import java.util.ArrayList;
import java.util.List;

import static toast.client.ToastClient.MODULE_MANAGER;

public class Panic extends Command {

    private final List<Module> wasEnabled = new ArrayList<>();

    public Panic() {
        super("Panic", ToastClient.cmdPrefix + "panic", "shutdowns client", false, "shutdown", "panic");
    }

    @Override
    public void run(String[] args) throws InterruptedException {
        if (mc.currentScreen != null) return;
        for (Module module : MODULE_MANAGER.getModules()) {
            if (module.isEnabled()) {
                module.setEnabled(false);
                wasEnabled.add(module);
            }
        }
        if (mc.inGameHud != null) {
            mc.inGameHud.getChatHud().clear(true);
            mc.updateWindowTitle();
        }
    }
}

File: src/main/java/toast/client/commands/cmds/Panic.kt
package toast.client.commands.cmds

import toast.client.ToastClient
import toast.client.commands.Command
import toast.client.modules.Module
import java.util.*

/**
 * Command to disable all modules and removes "all" traces of the client
 */
class Panic :
    Command("Panic", """${ToastClient.cmdPrefix}panic""", "shut's the client down'", false, "shutdown", "panic") {
    private val wasEnabled: MutableList<Module> = ArrayList()

    @Throws(InterruptedException::class)
    override fun run(args: Array<String>) {
        if (mc.currentScreen != null) return
        for (module in ToastClient.MODULE_MANAGER.modules) {
            if (module.enabled) {
                module.disable()
                wasEnabled.add(module)
            }
        }
        if (mc.inGameHud != null) {
            mc.inGameHud.chatHud.clear(true)
            mc.updateWindowTitle()
        }
    }
}